### PR TITLE
Bug fix/sparse graph register warning

### PR DESCRIPTION
**Pull request by mchacki**
Changed behaviour:
Wheneer we traverser over a non-existing vertex it was formerly printed in the Error log.
Now it is registered as a warning on the query.
In both cases the Query Continues and interprets the vertex as `null`
=> No apis will break. Information is just located at a better position, this thing is no error, but expected by ArangoDB, it might be unexpected by the users Datamodel.

@jsteemann @fceller 
In order to apply Mocks in UnitTests i needed to make Query and two functions virtual => has negative performance impact.
How should we treat this?

### CODE REVIEW

File: arangod/Aql/Query.h [hunk 1]
```diff
@@ -208,7 +208,7 @@ class Query {
   void releaseEngine();
 
   /// @brief return the transaction, if prepared
-  inline transaction::Methods* trx() { return _trx; }
+  virtual inline transaction::Methods* trx() { return _trx; }
```
Comment (jsteemann) [inline arangod/Aql/Query.h]: :+1: for `virtual inline`.

**Comment by jsteemann** [comment]
the two virtual functions should be ok. it may be sensible to change `virtual inline ...` to just `virtual ...` however.

File: arangod/Aql/Query.h (hunk 1 above)
Comment (fceller) [reply arangod/Aql/Query.h]: `virtual inline` will only make sense if you have an actual object at hand, not a pointer or reference.

Comment (mchacki) [reply arangod/Aql/Query.h]: ok we only have pointers afaik. Will fix

**Pull request merged**